### DISCUSSION TITLE
Mac Compiling Issue

### DISCUSSION THREAD

**Discussion by kylegrymonprez** [xLightsSequencer/xLights#4859] [General]
@dkulp are you getting compliation errors on mac using Metal? I've had to for OpenGL to get it to compile.

XCode 16.0
Macbook Pro M1 Max
Sonoma 14.6.1

Additionally, are my submodules messed up with FFMpeg, or is there something else more systemic.

Thanks!

**Comment by dkulp** [comment]
Seems to be compiling fine for me (and in Xcode cloud).     That said, both are using Sequoia.    

What errors are you seeing?

**Comment by kylegrymonprez** [comment]
If I don't go into xlGraphicsBase.h and `#define FORCE_OPENGL_BASE` then I get the following:
<img width="363" alt="Screenshot 2024-10-04 at 8 18 27 AM" src="https://github.com/user-attachments/assets/70e6d117-ca13-4a3b-a172-f5a257df55f3">
<img width="1033" alt="Screenshot 2024-10-04 at 8 18 36 AM" src="https://github.com/user-attachments/assets/ac4f3d0d-6aec-43b9-a2e9-6d8b878145c3">

I am able to get around it, I'm just wondering what I'm missing. 

Additionally, if I dont manually install the nanosvg dependencies, the XCode project wont work out of the gate. Maybe I'm messed up the depencies seetup somewhere so I'm still double checking. 

If it's a me issue, maybe I'm misunderstanding something in the setup process that I can work on cleaning up the readme's across the repositories.

**Comment by kylegrymonprez** [comment]
This might be a git-lfs thing.......running traces now show that it's just slow and could be a me issue - which is actually ideal. I'm going to let this bake and see if it fixes my issue......would explain a lot.....

**Comment by kylegrymonprez** [comment]
Okay - no still an issue after I let git-lfs run o the dependency directory for like.....30 minutes. How can I tell if I have the exact version XCode as the nightlies in order to see if I can use the dependency directory correctly? Otherwise I'll re-step through building all the dependencies on my own.

**Comment by dkulp** [comment]
It sounds like your xLights/macOS  is out of date.   Do a "git submodule update --init" in the root xLights directory to make sure all the submodules get updated.

**Comment by kylegrymonprez** [comment]
I think you're right.......i keep forgetting it's like 3 different projects to keep updating. I think I'm off and kicking again.
